### PR TITLE
Support new ZADD options introduced at Redis 3.0.2

### LINKED ISSUE #1067: New ZADD Options
Redis recently added new options to the [ZADD](http://redis.io/commands/ZADD) command as of 3.0.2.  Is anyone already planning on adding these?  I took a look at it, but it gets a bit hairy because the INCR option changes the return data type... Is it desired to add a new method interface for every combination of options?

### PR DESCRIPTION
**Pull request by HeartSaVioR**
Closes #1067 

We discussed many things from #1067, so please refer that issue for more details.

I didn't add javadoc for added methods for some reasons.  
I'll address what thing prevents me to add javadoc to #978.

Please review and comment. Thanks!

### CODE REVIEW

**Comment by jamespedwards42** [comment]
Looks great.  Thanks for handling all of this, sorry that it had to turn into such a fuss...  Looking forward to the *Param args as well.

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 1]
```diff
@@ -399,15 +401,15 @@ public void zadd(final byte[] key, final double score, final byte[] member) {
     sendCommand(ZADD, key, toByteArray(score), member);
   }
 
-  public void zaddBinary(final byte[] key, final Map<byte[], Double> scoreMembers) {
+  public void zadd(final byte[] key, final double score, final byte[] member,
+      final ZAddParams params) {
+    sendCommand(ZADD, params.getByteParams(key, toByteArray(score), member));
+  }
 
+  public void zaddBinary(final byte[] key, final Map<byte[], Double> scoreMembers) {
```
Comment (marcosnils) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: I know it has nothing to do with this PR. But I was looking at the `zaddBinary` method and I was thinking: "why is it called zaddBinary if zadd is already binary?". Then I realized that the difference is that `zaddBinary` accepts multiple score members. 

Maybe we can take the opportunity to rename this method to just `zadd`?. In Client.java the method that calls `zaddBinary` is called `zadd`.

**Comment by marcosnils** [comment]
@HeartSaVioR PR is perfect. Let's see what we want to do with the `zaddBinary` thing and then merge :)

File: src/main/java/redis/clients/jedis/BinaryClient.java (hunk 1 above)
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: @marcosnils 
Great finding! Renamed zaddBinary to zadd.
Maybe it was just a workaround if Java compiler or JVM could not distinguish methods if only param's Generic is different.
For now, all tests passed.

**Comment by marcosnils** [comment]
@HeartSaVioR LAGTM (Looks amazingly good to me??) :+1:

**Comment by HeartSaVioR** [comment]
@jamespedwards42 You don't need to be sorry. It was me who can't stand with that decision. :)

**Comment by HeartSaVioR** [comment]
Merged into master and 2.8 branch respectively.
Thanks all!

**Pull request merged**